src/main/java/org/opensearch/security/dlic/rest/api/RollbackVersionApiAction.java

```diff
@@ -0,0 +1,395 @@
+    }
+
+    private ValidationResult<SecurityConfiguration> rollbackCommon(String versionId, SecurityConfigVersionDocument doc) throws IOException {
+        var versionSpecificDoc = doc.getVersions().stream().filter(v -> versionId.equals(v.getVersion_id())).findFirst().orElse(null);
+
+        try {
+            rollbackConfigsToSecurityIndex(versionSpecificDoc);
+
+            return ValidationResult.error(OK, (builder, params) -> {
+                XContentBuilder inner = buildRollbackResponseJson(versionId);
+                builder.copyCurrentStructure(JsonXContent.jsonXContent.createParser(null, null, BytesReference.bytes(inner).streamInput()));
+                return builder;
+            });
+
+        } catch (Exception e) {
+            log.error("Rollback to version {} failed", versionId, e);
+            return ValidationResult.error(INTERNAL_SERVER_ERROR, payload(INTERNAL_SERVER_ERROR, "Rollback failed: " + e.getMessage()));
+        }
+    }
+
+    private void rollbackConfigsToSecurityIndex(SecurityConfigVersionDocument.Version<?> versionData) throws IOException {
+        Map<String, SecurityConfigVersionDocument.HistoricSecurityConfig<?>> securityConfigs = versionData.getSecurity_configs();
+        if (securityConfigs == null || securityConfigs.isEmpty()) {
+            throw new NullPointerException("No security configs to rollback in version " + versionData.getVersion_id());
+        }
+
+        Map<CType<?>, SecurityDynamicConfiguration<?>> configsToApply = new HashMap<>();
+        Map<CType<?>, SecurityDynamicConfiguration<?>> backups = new HashMap<>();
+
+        try {
+
+            ConfigurationMap currentConfigs = configRepository.getConfigurationsFromIndex(CType.values(), false, true);
+
+            for (Map.Entry<String, SecurityConfigVersionDocument.HistoricSecurityConfig<?>> entry : securityConfigs.entrySet()) {
+                String cTypeName = entry.getKey();
+                SecurityConfigVersionDocument.HistoricSecurityConfig<?> sc = entry.getValue();
+
+                CType<?> cType = CType.fromString(cTypeName);
+                if (cType == null) {
+                    throw new NullPointerException("Rollback aborted: Unknown config type '" + cTypeName + "' found in version");
+                }
+
+                if (sc == null || sc.getConfigData() == null) {
+                    log.warn("Skipping cType '{}' due to null configData", cTypeName);
+                    continue;
+                }
+
+                SecurityDynamicConfiguration<?> currentConfig = currentConfigs.get(cType);
+                if (currentConfig == null) {
+                    throw new IllegalArgumentException("Rollback aborted: Could not fetch current config for cType '" + cTypeName + "'");
+                }
+
+                SecurityDynamicConfiguration<?> rollBackConfig = SecurityDynamicConfiguration.empty(cType);
+                rollBackConfig.setSeqNo(currentConfig.getSeqNo());
```

Comment:
Are we allowed to overwrite this? My understanding is that this number only increases as modifications are made to a doc.